Mathlib/GroupTheory/GroupAction/SubMulAction.lean

```diff
@@ -421,3 +422,35 @@ theorem smul_mem_iff (s0 : s ≠ 0) : s • x ∈ p ↔ x ∈ p :=
 #align sub_mul_action.smul_mem_iff SubMulAction.smul_mem_iff
 
 end SubMulAction
+
+namespace SubMulAction
+
+/- The inclusion of a SubMulaction, as an equivariant map -/
+variable {M α : Type*} [Monoid M] [MulAction M α]
+
+
+/-- The inclusion of a SubMulAction into the ambient set, as an equivariant map -/
+def inclusion (s : SubMulAction M α) : s →[M] α where
+-- The inclusion map of the inclusion of a SubMulAction
+  toFun := fun x => x.val
+-- The commutation property
+  map_smul' _ _ := rfl
+
+theorem inclusion.toFun_eq_coe (s : SubMulAction M α) :
+    s.inclusion.toFun = fun x => x.val := rfl
+
+lemma image_inclusion (s : SubMulAction M α) :
+    Set.range s.inclusion = s.carrier  := by
+  ext a
+  simp only [Set.mem_range, Subtype.exists, mem_carrier, SetLike.mem_coe]
+  constructor
+  · rintro ⟨a, h, rfl⟩; exact h
+  · exact fun h ↦ ⟨a, h, rfl⟩
```

Comment:
Can you rewrite with the lemma above, and then use a lemma about the range of `Subtype.val`?

Reply:
Yes, but I used another lemma (that says ⇑s.inclusion = Subtype.val )